File: p4/Deepspace/BetaPowerEfficientSpaceStation.rb
#encoding:utf-8
require_relative "PowerEfficientSpaceStation.rb"
require_relative "BetaPowerEfficientSpaceStationToUI.rb"
require_relative "Dice.rb"
module Deepspace
    class BetaPowerEfficientSpaceStation < PowerEfficientSpaceStation
        @@EXTRAEFFICIENCY = 1.2 
        def initialize(station)
            super(station)
            @dice = Dice.new()
        end
        def fire()
            if dice.extraEfficiency()
                super*@@EXTRAEFFICIENCY
            else 
                super
            end 
        end
        def getUIversion()
			Deepspace::BetaPowerEfficientSpaceStationToUI.new(self)
		end 
        def to_s
            "(Beta)"+super
        end 
    end
    #station = SpaceStation.new("p",SuppliesPackage.new(1,2,3))
    #puts station.fire()
    #puts station.protection()
    #bpes = BetaPowerEfficientSpaceStation.new(station)
    #puts bpes.fire()
    #puts bpes.protection()
end

File: p4/deepspace/src/deepspace/BetaPowerEfficientSpaceStation.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

import java.util.ArrayList;

/**
 *
 * @author Sergio
 */
public class BetaPowerEfficientSpaceStation extends PowerEfficientSpaceStation{
    
    private Dice dice;
    private static final float EXTRAEFFICIENCY = 1.20f;
    
    BetaPowerEfficientSpaceStation(SpaceStation station){                     
        super(station);  
        dice = new Dice();
    }
    
    @Override
    public float fire(){
        if(dice.extraEfficiency()){           
            return 0;
        }
        else{
            return super.fire();
        }
    }
    
    public BetaPowerEfficientSpaceStationToUI getUIversion(){
        return new BetaPowerEfficientSpaceStationToUI(this);
    }
}
